### PR TITLE
add fiat client

### PR DESCRIPTION
**Pull request by yasmin-abdullah**
Add fiat client with authorize user endpoint.

### CODE REVIEW

File: pkg/fiat/client.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func NewClient(url string) Client {
+	return &client{
+		url: url,
+	}
+}
+
+func NewDefaultClient() Client {
+	return &client{
```
Comment (billiford) [inline pkg/fiat/client.go:22]: Change the return for the default client to `return NewClient("http://spin-fiat.spinnaker:7003")`. Maybe set the default URL as a const if you want.

File: pkg/fiat/client_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	})
+
+	Describe("#NewDefaultClient", func() {
+		BeforeEach(func() {
+			client = NewDefaultClient()
+		})
+
+		//This does nothing, ask Billy why it was added to othe clients?
+		It("succeeds", func() {
+		})
```
Comment (billiford) [inline pkg/fiat/client_test.go:39]: This is just a cheat for test coverage.

**Review by billiford — CHANGES_REQUESTED** [review]
See comment for new default client.

File: pkg/fiat/client.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package fiat
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const fiatUrl = "http://localhost:7003"
```
Comment (billiford) [inline pkg/fiat/client.go:10]: Shouldn't this be `http://spin-fiat.spinnaker:7003`?

Comment (yasmin-abdullah) [reply pkg/fiat/client.go:10]: maybe, I was cheating from the arcade client https://github.com/billiford/go-clouddriver/blob/master/pkg/arcade/client.go#L24

Comment (billiford) [reply pkg/fiat/client.go:10]: Ah yeah, arcade is running in the same pod as go-cd so its default will be `localhost`. The default for fiat will be what I posted above.

**Review by billiford — APPROVED** [review]
LGTM!

**Pull request merged**